Docs: create component page (#168)
BrightspaceUI/documentation: docs/adding-component.md

- [Preparing the component](#preparing-the-component) (in the component's repo)
- [Creating the component issue](#creating-the-component-issue) (in this repo)

## Preparing the Component

### Component Code

Add JSDoc comments to your component. See [input-text](https://github.com/BrightspaceUI/core/blob/master/components/inputs/input-text.js) as an example. Include each of the following where applicable:
- Description
- `@slot` (slots)
- `@fires` (events)
- Properties descriptions and `@type` if needed (see `type` property in `input-text`)

### Generating custom-elements.json

The `custom-elements.json` file is used for the component's properties, slots, and events.

1. Ensure the `release.yml` GitHub action has a step that generates `custom-elements.json`. For example:
```
- name: Create custom-elements.json
  run: npx wca analyze \"src/*.js\" --format json --outFile custom-elements.json
```
2. In `package.json` add `custom-elements.json` to `"files"` array.

### README/Markdown Document

1. Remove `properties`, `slots`, and `events` sections. Rename `Accessibility` section to `Accessible Properties`.
2. Order the document as follows:
- Title
- Description

> - Demo with code hidden that displays all components within that page (e.g., for Button page would should all button versions such as regular, subtle, icon). See [Code Blocks](#code-blocks).

Reviewer: ```suggestion
- Demo with code hidden that displays a preview of all components described within that page (e.g., for the Button page  it would show all button versions such as regular, subtle, icon). See [Code Blocks](#code-blocks).
```

- Design content (could include Best Practices (see [Best Practices](#best-practices)), Accessibility, and Responsive Behaviour)
- Component heading with tag beside it (e.g., `## Button [d2l-button]`)
- Live demo (see [Code Blocks](#code-blocks))
- Any other content that should appear in the document, potentially also including a secondary demo or other code block (see [Code Blocks](#code-blocks)).
- Content that should not appear in the Daylight site (see [Hidden Content](#hidden-content))
3. When these changes are merged, ensure a release happens in order for the Daylight site to pick up the changes.

#### Best Practices

The content of the best practices section should be formatted as:
```
<!-- docs: start best practices -->
<!-- docs: start dos -->
* Do this
<!-- docs: end dos -->

<!-- docs: start donts -->
* Don't do that
<!-- docs: end donts -->
<!-- docs: end best practices -->
```

#### Code Blocks

**Code-hidden demo:** This demo shows only the visual portion of the component(s) without the code. It is generally used at the top of a component page.

Add the comment `<!-- docs: demo -->` within the code block. For example:
```
```html

> <!-- docs: demo -->

Reviewer: Should we show size options on all of the examples?

## Creating the Component Issue

### Requested Component

If you have a component idea that you would like to suggest then you can [submit a GitHub issue](https://github.com/BrightspaceUI/documentation/issues/new?assignees=&labels=Requested+Component&template=component-request.md&title=%3CComponent+Name%3E) for it. These component requests will appear in the Daylight site in the status pages, and will have pages generated from their issues.

### Labs Component

Labs components generally do not have complete documentation pages in the Daylight site, but do appear in the Status table and also have pages generated from their issues. To document these, [use the GitHub issue template](https://github.com/BrightspaceUI/documentation/issues/new?assignees=&labels=Labs+Component&template=component-in-progress.md&title=%3CComponent+Name%3E).

### Official Component

Once the [Preparing the Component](#preparing-the-component) steps have been completed, [open a GitHub issue](https://github.com/BrightspaceUI/documentation/issues/new?assignees=&labels=Official+Component&template=component-documented.md&title=%3CComponent+Name%3E).

Notes:

> - Open the issue then close it if documentation and development are complete.

Reviewer: ```suggestion
- Issues on the documentation repository should only be closed if documentation and development are complete.
```

- If the component is ready for the production site, add the `Published` label. This will open a PR so that you can preview the component's page.

> - `eleventyNavigation`:

Reviewer: Should this be here? It seems like this may be better off being described within the issue template
